typing_extensions/src_py3/typing_extensions.py

```diff
@@ -2252,6 +2252,12 @@ def __repr__(self):
                 prefix = '~'
             return prefix + self.__name__
 
+        def __hash__(self):
+            return hash((self.__name__,
```

Comment:
I think you can just do `object.__hash__(self)`. TypeVars are also hashed by identity:
```
>>> T1 = TypeVar("T")
>>> T2 = TypeVar("T")
>>> hash(T1), hash(T2)
(283358494, 283358500)
```